src/query/sql/src/planner/optimizer/cascades/cascade.rs

```diff
@@ -29,36 +30,36 @@ use crate::optimizer::cost::DefaultCostModel;
 use crate::optimizer::format::display_memo;
 use crate::optimizer::memo::Memo;
 use crate::optimizer::rule::TransformResult;
-use crate::optimizer::RuleSet;
 use crate::optimizer::SExpr;
 use crate::IndexType;
-use crate::MetadataRef;
 
 /// A cascades-style search engine to enumerate possible alternations of a relational expression and
 /// find the optimal one.
 pub struct CascadesOptimizer {
-    pub(crate) memo: Memo,
-    pub(crate) cost_model: Box<dyn CostModel>,
+    pub memo: Memo,
+    pub explore_rule_set: roaring::RoaringBitmap,
```

Comment:
I think we should not to expose the `RoaringBitmap` here, `RuleSet`  is better.

Reply:
> I think we'd better not expose the `RoaringBitmap` here, should use `RuleSet` instead.

I understand what you mean, I want to transform the RuleSet to use bitmap management, and then submit a PR. If there is no problem with this PR, can it be passed first？

Reply:
> If there is no problem with this PR, can it be passed first？

Please address it in this PR.